### PR TITLE
Fix artifact analyzers to consume power only during scanning

### PR DESCRIPTION
**Pull request by spacedwarf14**
## About the PR

Currently artifact analyzers consume 12000W regardless of whether they are in use or not. In the YML it is commented that the intention is to only consume power during scans, but this is not the case because the APC load doesn't get updated. I think there was confusion about what apc.NeedsPower actually does, as it shouldn't affect the receiver's load (but rather, the receiver's requirement for power in the first place).

Technically wattage is set to 1 while not scanning for upstream APC bugs reasons, as @dvir001 mentioned earlier.

## Why / Balance

It would certainly reduce power / fuel requirements for science ships. However the same effect is achieved by unanchoring the analyzer. Currently it's just an extra step that feels lrp.

## How to test

Spawn a science ship, place artifact and scan / finish scan while monitoring power. Experiment with toggling power during scans as well to make sure pause / resume logic wasn't affected.

## Media

2024-09-23-162234_586x750_scrot
2024-09-23-162219_653x819_scrot

## Requirements

- [X] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

Not a breaking change, but something I'd like to do better before this is merged: when the analyzer begins scanning, the sudden 0->12kW power draw causes apc brownout. This
1) closes the analyze console UI. not the friendliest effect.
2) Emits a PowerChanged event, which then pauses the scan, then is issued again once the APC is ramped up to resume the scan. Pausing and resume the scan doesn't call  SetPowerSwitch(...) because it's actually only called when power is lost and restored, and only concerns the ActiveArtifactAnalyzerComponent. Anyway, it's kinda noisy and waste of cpu cycles.

ideas?

**Changelog**

:cl:
- fix: Artifact analyzers do not consume power when idle

### CODE REVIEW

**Comment by spacedwarf14** [comment]
Maybe there's a way for a power receiver to ramp up its own consumption (just as a generator does with its supply power).

**Comment by dvir001** [comment]
Based

File: Resources/Prototypes/Entities/Structures/Machines/artifact_analyzer.yml [hunk 1]
```diff
@@ -49,7 +49,7 @@
     anchored: true
   - type: ApcPowerReceiver
     powerLoad: 12000
-    needsPower: false #only turns on when scanning
+    needsPower: true
```
Comment (dvir001) [inline Resources/Prototypes/Entities/Structures/Machines/artifact_analyzer.yml:52]: Frontier comment missing

**Comment by Cheackraze** [comment]
the power ramping on the back end definitely leads to this, but I see it more as a shortcoming of the underlying power ramping system not being ideal.

File: Content.Server/Xenoarchaeology/Equipment/Systems/ArtifactAnalyzerSystem.cs [hunk 2]
```diff
@@ -491,16 +497,26 @@ private void OnItemRemoved(EntityUid uid, ArtifactAnalyzerComponent component, r
 
     private void OnAnalyzeStart(EntityUid uid, ActiveArtifactAnalyzerComponent component, ComponentStartup args)
     {
-        if (TryComp<ApcPowerReceiverComponent>(uid, out var powa))
-            powa.NeedsPower = true;
+        if (!TryComp<ApcPowerReceiverComponent>(uid, out var powa))
```
Comment (dvir001) [inline Content.Server/Xenoarchaeology/Equipment/Systems/ArtifactAnalyzerSystem.cs:500]: Frontier comment kind of missing

**Review by whatston3 — APPROVED** [review]
Made a few changes for consistent commenting to Frontier changes.

Looks good, works fine under test.

File: Content.Server/Xenoarchaeology/Equipment/Systems/ArtifactAnalyzerSystem.cs (hunk 2 above)
Comment (spacedwarf14) [reply Content.Server/Xenoarchaeology/Equipment/Systems/ArtifactAnalyzerSystem.cs:504]: Thanks, still learning the ropes, seems like your commits address the comments so marking as resolved.

**Pull request merged**